src/controller/python/test/unit_tests/test_yaml_format_converter.py

```diff
@@ -0,0 +1,52 @@
+#
+#    Copyright (c) 2022 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+from chip.yaml.format_converter import convert_yaml_octet_string_to_bytes
+from binascii import unhexlify
+import unittest
+
+
+class TestOctetStringYamlDecode(unittest.TestCase):
+    def test_common_cases(self):
+        self.assertEqual(convert_yaml_octet_string_to_bytes("hex:aa55"), unhexlify("aa55"))
+        self.assertEqual(convert_yaml_octet_string_to_bytes("hex:"), unhexlify(""))
+        self.assertEqual(convert_yaml_octet_string_to_bytes("hex:AA55"), unhexlify("aa55"))
+
+        self.assertEqual(convert_yaml_octet_string_to_bytes("0\xaa\x55"), unhexlify("30aa55"))
+        self.assertEqual(convert_yaml_octet_string_to_bytes("0\xAA\x55"), unhexlify("30aa55"))
+        self.assertEqual(convert_yaml_octet_string_to_bytes("0\xAa\x55"), unhexlify("30aa55"))
+
+        self.assertEqual(convert_yaml_octet_string_to_bytes("0hex:"), b"0hex:")
+        self.assertEqual(convert_yaml_octet_string_to_bytes("0hex:A"), b"0hex:A")
+        self.assertEqual(convert_yaml_octet_string_to_bytes("0hex:AA55"), b"0hex:AA55")
+
+        self.assertEqual(convert_yaml_octet_string_to_bytes("AA55"), b"AA55")
+        self.assertEqual(convert_yaml_octet_string_to_bytes("AA\n\r\t55"), unhexlify("41410a0d093535"))
+        # TODO(#23669): After utf8 is properly supported expected result is unhexlify("c3a9c3a90a0a")
+        self.assertEqual(convert_yaml_octet_string_to_bytes("\xC3\xA9é\n\n"), unhexlify("c3a9e90a0a"))
```

Comment:
We should just not allow non-ASCII chars in byte strings, imo.  Too much confusion in terms of _which bytes_ they represent (e.g. why would it be UTF-8?  Right now in YAML it comes down to "convert to UTF-16 and then take the low byte of each code unit" or something insane like that.)

Reply:
Yes that is what is happening today.

My objective with this PR is to bring python parsing up to parity of javascript. I think the effort on enforcing not allowing non-ASCII chars in byte strings, or whether we should properly support UTF-8 is something that needs to apply to both and will be handled in a follow up PR resolving issue https://github.com/project-chip/connectedhomeip/issues/23669